src/controller/java/src/matter/controller/Messages.kt

```diff
@@ -0,0 +1,206 @@
+   */
+  class Event(
+    val path: EventPath,
+    val eventNumber: ULong,
+    val priorityLevel: UByte,
+    val timeStamp: Timestamp,
+    val data: ByteArray
+  ) : ReadData()
+
+  /**
+   * Represents data related to an attribute.
+   *
+   * @param path The attribute path associated with the data.
+   * @param data The ByteArray containing the data in TLV format.
+   */
+  class Attribute(val path: AttributePath, val data: ByteArray) : ReadData()
+}
+
+/** Represents a failure that can occur during a read operation. */
+sealed class ReadFailure {
+  /**
+   * Represents a failure related to an event path.
+   *
+   * @param path The event path information associated with the failure.
+   * @param error The exception that describes the failure.
+   */
+  class Event(val path: EventPath, val error: Exception) : ReadFailure()
+
+  /**
+   * Represents a failure related to an attribute path.
+   *
+   * @param path The attribute path information associated with the failure.
+   * @param error The exception that describes the failure.
+   */
+  class Attribute(val path: AttributePath, val error: Exception) : ReadFailure()
+}
+
+/**
+ * Represents the response from a read operation, containing both successes and failures.
+ *
+ * @param successes A list of successfully read data elements.
+ * @param failures A list of failures that occurred during the read operation.
+ */
+class ReadResponse(val successes: List<ReadData>, val failures: List<ReadFailure>)
+
+/**
+ * Information about a subscribe request element.
+ *
+ * @param eventPaths A list of event path information in the read request.
+ * @param attributePaths A list of attribute path information in the read request.
+ * @param minInterval The minimum interval boundary floor in seconds.
+ * @param maxInterval The maximum interval boundary ceiling in seconds.
+ * @param keepSubscriptions Indicates whether to keep existing subscriptions.
+ * @param fabricFiltered Limits the data read within fabric-scoped lists to the accessing fabric.
+ */
+class SubscribeRequest(
+  val eventPaths: List<EventPath>,
+  val attributePaths: List<AttributePath>,
+  val minInterval: Duration = Duration.ofSeconds(SUBSCRIPTION_MIN_INTERVAL_S),
+  val maxInterval: Duration = Duration.ofSeconds(SUBSCRIPTION_MAX_INTERVAL_S),
+  val keepSubscriptions: Boolean = true,
+  val fabricFiltered: Boolean = true
+)
+
+/** An interface representing the possible states of a subscription. */
+sealed interface SubscriptionState {
+  /**
+   * Represents an error notification in the subscription.
+   *
+   * @param terminationCause The cause of the subscription termination.
+   */
+  class SubscriptionErrorNotification(val terminationCause: UInt) : SubscriptionState
+
+  /**
+   * Represents an update in the state of a subscribed node.
+   *
+   * @param updateState The state update received from the subscribed node.
+   */
+  class NodeStateUpdate(val updateState: ReadResponse) : SubscriptionState
```

Comment:
Really, we should rename ReadResponse to ReportDataResponse, from spec, ReportDataResponse is used by both Read and Subscribe interaction as response.

Reply:
"Yes, the 'subscribe' function returns the same result as 'read.' The Kotlin API does not strictly adhere to the interface message. Here is the prototype for the 'read' function we decided to use from the API review:

`suspend fun read(request: ReadRequest): ReadResponse`

This alignment makes 'ReadResponse' better matched with 'ReadRequest'."

Reply:
I think read symmetry is nice. It leaks here a bit since we probably would find `SubscribeResponse` to be natural. We could `type SubscribeState = ReadResponse` however that seems to just create noise. Spec says they are the same, so if we keep them the same maybe some code can be shared.